compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java

```diff
@@ -1674,6 +1674,12 @@ public BType readType(int cpI) throws IOException {
                     if (Symbols.isFlagOn(flags, Flags.ANONYMOUS)) {
                         objectSymbol.flags |= Flags.ANONYMOUS;
                     }
+                    if (Symbols.isFlagOn(flags, Flags.DISTINCT)) {
```

Comment:
Can you add a few tests that loaded through BIR and check for type relationships?
eg:

```
import ballerina/io;

public type Foo distinct isolated client object {
    isolated function execute() returns anydata|error;
};

public type Bar distinct isolated client object {
    isolated function execute() returns anydata|error;
};

public function test(Foo foo, Bar bar) {
    io:println(foo is Bar); //incompatible types: 'Foo' will not be matched to 'Bar'
}
```

Reply:
Added.